### PR TITLE
Fix #25633: Favorite Pages Add Scroll To Top Feature

### LINKED ISSUE #25633: Favorite Pages: Lacks 'Go to Top' scroll on changing pages using pagination
### Parent Issue

[Pages]

### Problem Statement

When changing pages using pagination in the 'Pages' portlet, there is no 'Go to Top' scroll feature. 

### Steps to Reproduce

1. Go to the 'Pages' portlet
2. Use the pagination feature to navigate to a different page
3. Observe that there is no 'Go to Top' scroll feature

### Acceptance Criteria

Upon changing pages using the pagination feature in the 'Pages' portlet, a 'Go to Top' scroll feature should be present to facilitate user navigation.

### dotCMS Version

23.07

### Proposed Objective

User Experience

### Proposed Priority

Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

https://github.com/dotCMS/core/assets/129882031/a4ae5074-bf39-47e4-a8a1-03d4b8eb171f

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by zJaaal**
### Summary
📜🐛♻️

This pull request fixes a bug and enhances the user experience of the dot-pages listing panel. It adjusts the scroll element type and logic, and adds a `scrollToTop` method that is called when the user changes the page of the `dot-listing-data-table` component.

> _Oh we are the coders of the `DotPagesListingPanel`_
> _We fix the bugs and make the code more swell_
> _We heave and we ho and we scroll to the top_
> _When we change the page of the `dot-listing-data-table`_

### Walkthrough
*  Add `scrollToTop` method to `DotPagesListingPanelComponent` class to scroll the container element to the top position ([link](https://github.com/dotCMS/core/pull/25724/files?diff=unified&w=0#diff-68cb42cd0420b2fd6a9d2f4305173ae7d75a2fd24205f2ec9bded9e298eb8784R190-R196))
*  Use `scrollToTop` method in `onPage` event handler of `dot-listing-data-table` component and `dotSiteService.switchSite$` observable subscription to reset the scroll position when navigating between pages of the table or switching sites ([link](https://github.com/dotCMS/core/pull/25724/files?diff=unified&w=0#diff-1dc91e554b57df9862b7aa55f54839c49ea15c3a4caa1bf096f96e5df2e97c4eR22), [link](https://github.com/dotCMS/core/pull/25724/files?diff=unified&w=0#diff-68cb42cd0420b2fd6a9d2f4305173ae7d75a2fd24205f2ec9bded9e298eb8784L81-R81))
*  Fix bug where scroll event listener was not attached to the correct element by using `document.querySelector` instead of `table.el.nativeElement.querySelector` to assign `scrollElement` property ([link](https://github.com/dotCMS/core/pull/25724/files?diff=unified&w=0#diff-68cb42cd0420b2fd6a9d2f4305173ae7d75a2fd24205f2ec9bded9e298eb8784L72-R74))
*  Prevent runtime errors by using optional chaining operator (`?.`) to safely access `scrollElement` property in `ngOnDestroy` method and `scrollToTop` method ([link](https://github.com/dotCMS/core/pull/25724/files?diff=unified&w=0#diff-68cb42cd0420b2fd6a9d2f4305173ae7d75a2fd24205f2ec9bded9e298eb8784L40-R40), [link](https://github.com/dotCMS/core/pull/25724/files?diff=unified&w=0#diff-68cb42cd0420b2fd6a9d2f4305173ae7d75a2fd24205f2ec9bded9e298eb8784L88-R88))

## Screenshots

https://github.com/dotCMS/core/assets/63567962/8faaeb4b-3b53-4dbc-9da3-5d648c4350c9

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-listing-panel/dot-pages-listing-panel.component.ts [hunk 1]
```diff
@@ -187,4 +187,11 @@ export class DotPagesListingPanelComponent implements OnInit, OnDestroy, AfterVi
         this.store.getPages({ offset: 0 });
         this.store.setSessionStorageFilterParams();
     }
+
+    scrollToTop(): void {
+        this.scrollElement?.scroll({
+            top: 0,
+            left: 0
+        });
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-listing-panel/dot-pages-listing-panel.component.ts:195]: Why this is not happeing on the level where `dot-pages` lives? We can outoput the event here or even listen in the store if there is a page change (I think)

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-listing-panel/dot-pages-listing-panel.component.ts (hunk 1 above)
Comment (zJaaal) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-listing-panel/dot-pages-listing-panel.component.ts:195]: We need to reset the scroll when a site switch happens and also we need to close the context menu from the favorite panel when a scroll happens.

```
 this.scrollElement = document.querySelector('dot-pages');

        this.scrollElement?.addEventListener('scroll', () => {
            this.closeContextMenu();
            this.tableScroll.emit();
        });

        this.dotSiteService.switchSite$.pipe(takeUntil(this.destroy$), skip(1)).subscribe(() => {
            this.store.getPages({ offset: 0 });
            this.scrollToTop(); // To reset the scroll so it shows the data it retrieves
        });
```

Comment (zJaaal) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-listing-panel/dot-pages-listing-panel.component.ts:195]: I think I can move this logic to `dot-pages` and emit a page change, but I'm not sure of how to close the Table context menu on scroll.

**Review by oidacra — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages.component.spec.ts [hunk 2]
```diff
@@ -353,6 +374,13 @@ describe('DotPagesComponent', () => {
         });
     });
 
+    it('should call scrollToTop method from DotPagesListingPanel', () => {
+        const elem = de.query(By.css('dot-pages-listing-panel'));
```
Comment (manuelrojas) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages.component.spec.ts:378]: can we use data-testId?

**Review by manuelrojas — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages.component.spec.ts (hunk 2 above)
Comment (zJaaal) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages.component.spec.ts]: done!

**Review by fmontes — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Pull request merged**